src/transformStateWithClones.js

```diff
@@ -7,7 +7,42 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let stateCopy = { ...state };
+
+  for (let i = 0; i < actions.length; i++) {
+    let actionsType;
+
+    switch (actions[i].type) {
+      case 'addProperties':
+        actionsType = actions[i].extraData;
+
+        result.push(Object.assign(stateCopy, actionsType));
+        break;
+
+      case 'removeProperties':
+        actionsType = actions[i].keysToRemove;
+
+        for (const key of actionsType) {
+          delete stateCopy[key];
+        }
+
+        result.push(stateCopy);
+        break;
+
+      case 'clear':
+        for (const key in stateCopy) {
+          delete stateCopy[key];
+        }
+
+        result.push(stateCopy);
+        break;
+    }
```

Comment:
What if type will have some other value?

Switch/case construction should have `default` case